appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -6,16 +6,23 @@
 import { StringDictionary } from 'azure-arm-website/lib/models';
 import { AzureParentTreeItem, AzureTreeItem, IActionContext, ICreateChildImplContext, TreeItemIconPath } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
+import { SiteClient } from "../SiteClient";
 import { AppSettingTreeItem } from './AppSettingTreeItem';
 import { getThemedIconPath } from './IconPath';
 import { ISiteTreeRoot } from './ISiteTreeRoot';
 
-export function validateAppSettingKey(settings: StringDictionary, newKey?: string, oldKey?: string): string | undefined {
-    newKey = newKey ? newKey.trim() : '';
-    oldKey = oldKey ? oldKey.trim().toLowerCase() : oldKey;
+export function validateAppSettingKey(settings: StringDictionary, client: SiteClient, newKey?: string, oldKey?: string): string | undefined {
+    newKey = newKey ? newKey : '';
+
+    if (client.isLinux && RegExp('[^\\w\\.]+').test(newKey)) {
+        return 'App setting names can only contain letters, numbers (0-9), periods ("."), and underscores ("_")';
+    }
     if (newKey.length === 0) {
-        return 'Key must have at least one non-whitespace character.';
+        return 'App setting names must have at least one character.';
```

Comment:
How come you removed "non-whitespace"? Can I have a key that is just whitespace? If so - it will be trimmed to nothing below which seems contradictory

Reply:
It is possible to have a whitespace only key in the portal. The trim below is for comparison with existing keys but somewhere down the line (outside of this function) we trim the keys anyway so I'll add the 'non-whitespace' bit back in